CMakeLists.txt

```diff
@@ -85,6 +85,15 @@ include_directories(
   ${CMAKE_CURRENT_SOURCE_DIR}/include
   ${CMAKE_CURRENT_BINARY_DIR}/include)
 
+############## Code Coverage #########################
+option(ENABLE_COVERAGE "Enable coverage reporting." OFF)
+
+if (ENABLE_COVERAGE)
+  include(CodeCoverage)
+  APPEND_COVERAGE_COMPILER_FLAGS()
+endif()
+
```

Comment:
So, I've tested, and I think I would actually move this to the root `CMakeLists.txt`, at the bottom after `add_subdirectory(tests)` is processed. The coverage report is generated at the top level anyway, not in the `tests/` subdirectory, and that way it keeps CMake from polluting `tests/` with target-tracking for the coverage target.

My commit b4682ac makes that change, and also adds some additional tooling to, among other things...
1. Turn `DISABLE_TESTS` into a full-fledged CMake `option()`
1. Check if the user has set `DISABLE_TESTS` and `ENABLE_COVERAGE` together, and emit a warning before overriding and force-enabling the tests
1. Check if `CMAKE_BUILD_TYPE` is **unset**, and if so automatically set it to `"Debug"`. (If it's something other than debug, the `WARNING` in `CodeCoverage.cmake` will fire, so no need to handle any of those cases.)

Reply:
Yeah, the previous comment was attached to the wrong block. Oops.